<a href="https://colab.research.google.com/github/uthours/Data-Processing-Visualization/blob/main/quiz19.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Visualization
# Dr. Jacob, ECU

# Check out the Air B&B dataset, https://raw.githubusercontent.com/nurfnick/Data_Viz/main/AB_NYC_2019.csv. Examine how many entries are null. Impute for atleast two columns that are null in an approriate fashion.

In [9]:
import pandas as pa

df = pa.read_csv("https://raw.githubusercontent.com/nurfnick/Data_Viz/main/AB_NYC_2019.csv") 
df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

# Let's examine  the number of null entries.

In [10]:
df.isna().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [11]:
df.isna().sum().sum() 

20141

There were total 20141 number of NaN or null entries throughout the entire dataframe

# let's find how many null value

In [12]:
dflatitude = pa.read_csv("https://raw.githubusercontent.com/nurfnick/Data_Viz/main/AB_NYC_2019.csv") 

dflatitude['latitude'].isna().sum()

0

In [13]:
dfneighbourhood_group = pa.read_csv("https://raw.githubusercontent.com/nurfnick/Data_Viz/main/AB_NYC_2019.csv") 

dfneighbourhood_group['neighbourhood_group'].isna().sum()

0

let's find how many null value are in name

In [14]:
dfname=pa.read_csv("https://raw.githubusercontent.com/nurfnick/Data_Viz/main/AB_NYC_2019.csv") 
dfname['name'].isna().sum()

16

Now, Let's get rid of thouse 16 null value.

In [15]:
dfname.fillna(value = 8888, #The fillna() method replaces the NULL values with any specified value, I am assigning 8888 in this case
          inplace =True)
dfname['name'].isna().sum() #this should give 0 as I have already assign null host_name some value (9999)

0